plugins/Capacity.py

```diff
@@ -0,0 +1,133 @@
+from modules.OsmoseTranslation import T_
+from plugins.Plugin import Plugin
+from plugins.Plugin import TestPluginCommon
+
+###########################################################################
+##                                                                       ##
+## Copyrights Éric Gillet 2020                                           ##
+##                                                                       ##
+## This program is free software: you can redistribute it and/or modify  ##
+## it under the terms of the GNU General Public License as published by  ##
+## the Free Software Foundation, either version 3 of the License, or     ##
+## (at your option) any later version.                                   ##
+##                                                                       ##
+## This program is distributed in the hope that it will be useful,       ##
+## but WITHOUT ANY WARRANTY; without even the implied warranty of        ##
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         ##
+## GNU General Public License for more details.                          ##
+##                                                                       ##
+## You should have received a copy of the GNU General Public License     ##
+## along with this program.  If not, see <http://www.gnu.org/licenses/>. ##
+##                                                                       ##
+###########################################################################
+
+
+class Capacity(Plugin):
+    def init(self, logger):
+        Plugin.init(self, logger)
+
+        self.errors[35001] = self.def_class(
+            item=3500,
+            level=2,
+            tags=["tag"],
+            title=T_("Invalid capacity value"),
+            detail=T_("""A capacity tag value is incorrect."""),
+            resource="https://wiki.openstreetmap.org/wiki/Key:capacity for more information",
+        )
+
+    def node(self, data, tags):
+        try:
+            if tags["capacity"] == "":
+                # Ignore errors that should be reported by generic prefix analysers
+                return
+            total_capacity = int(tags["capacity"])
+        except KeyError:
+            # No capacity tag
+            return
+        except ValueError:
+            return {
+                "class": 35001,
+                "subclass": 0,
+                "text": T_('Invalid "{0}" value: {1}', "capacity", tags["capacity"]),
+            }
```

Comment:
Maybe we can check this in Number plugin ?

Reply:
I thought about adding valid integer and min/max validation, but it would introduce additional complexity to Number, which seems already not really [straightforward](https://github.com/osm-fr/osmose-backend/blob/8709c8e/plugins/Number.py#L60-L74). Also it currently allows having an height defined in kilograms ^^.

I tried defining a [declarative rule set](https://gitlab.com/-/snippets/2038252) for the Number plugin (or any plugin really), but while rules would be more explicit and precise, that would also make it tricky to handle edge-cases.

In conclusion I think it would be easier and clearer to handle each ~numerical "tag family" (speed, height, capacity etc.) in its own plugin/class to avoid convoluted case handling, even if that causes a bit of repetition.

What do you think ?

Reply:
MapCSS may be more appropriate for "declarative" rule set. JOSM already have a MapCSS for this.

Reply:
JOSM does have some numeric validation in MapCSS, but not for `capacity`, `capacity:*`. Can you expand on what you think should be done ? Thank you :)

Reply:
The rule can (probably) also be written in MapCSS an shared with JOSM.

Reply:
I agree that declarative (and MapCSS should be preferred when possible. This particular error could indeed be handled by a MapCSS rule. However :

* The plugin must do this check anyway to compare the value to sub-capacities later. Why not raise the error ?
* [MapCSS don't support wildcard/regex for keys](https://wiki.openstreetmap.org/wiki/MapCSS/0.2#Tests), so the same check for subcapacities must be implemented in this plugin anyway.
* That would split error handling on this tag between two plugins.

Do you think I should make a separate MapCSS and Plugin containing this check only ? I don't mind as it seems easy enough thanks to [mapcss2osmose.py](https://github.com/osm-fr/osmose-backend/blob/master/mapcss/mapcss2osmose.py).

Reply:
So ok, we keep think like this.